src/App.js

```diff
@@ -28,12 +28,21 @@ import fish from './MockFish.js'
   
   render() {
     return (
-      
+    
 <Router>
   <Header/>
     <Switch>
     <Route exact path="/" component={Home} />
-    <Route path="/fishindex" component={FishIndex} />
+    <Route path="/fishindex" render={(props) => <FishIndex fish={this.state.fish}/>} />
+    
+          <Route
+            path="/fishshow/:id"
+            render={(props) => {
+              let id = props.match.params.id
+              let fish = this.state.fish.find(c => c.id === +id)
```

Comment:
This code looks good, but I would be more clear about your local variable in the `.find()` method. Maybe changing `c` to `f` ?

Reply:
fix in progress